article.md

```diff
@@ -0,0 +1,612 @@
+
+Let's define `DbInterface`.
+
+```typescript
+// src/typings/DbInterface/index.d.ts
+import * as Sequelize from "sequelize";
+import { CommentAttributes, CommentInstance } from "src/models/Comment";
+import { PostAttributes, PostInstance } from "src/models/Post";
+import { UserAttributes, UserInstance } from "src/models/User";
+
+export interface DbInterface {
+  sequelize: Sequelize.Sequelize;
+  Sequelize: Sequelize.SequelizeStatic;
+  Comment: Sequelize.Model<CommentInstance, CommentAttributes>;
+  Post: Sequelize.Model<PostInstance, PostAttributes>;
+  User: Sequelize.Model<UserInstance, UserAttributes>;
+}
+```
+
+Next, `models/index.ts`, where `createModels` lives.
+
+```typescript
+// src/models/index.ts
+import * as Sequelize from "sequelize";
+import { DbInterface } from "src/typings/DbInterface";
+import { UserFactory } from "./User";
+import { PostFactory } from "./Post";
+import { CommentFactory } from "./Comment";
+
+export const createModels = (databaseId: string, sequelizeConfig): DbInterface => {
+  const { url: dbConnectionString } = sequelizeConfig;
+  const sequelize = new Sequelize(dbConnectionString, sequelizeConfig);
+
+  const db: DbInterface = {
+    sequelize,
+    Sequelize,
+    Comment: CommentFactory(sequelize, Sequelize),
+    Post: PostFactory(sequelize, Sequelize),
+    User: UserFactory(sequelize, Sequelize)
+  };
+
+  return db;
+};
+```
+
+Woah, we can finally write our api routes! That didn't take too long. Let's do that.
+
+```typescript
+// src/app.ts
+app.get('/users', (req: Request, res: Response) => {
+  db.User.findAll()
+    .then((users: UserInstance[]) => res.status(200).json({ users }));
+    .catch(err => res.status(500).json({ err: 'oops' }));
+});
+
+app.get('/comments', (req: Request, res: Response) => {
+  db.Comment.findAll()
+    .then((comments: CommentInstance[]) => res.status(200).json({ comments }));
+    .catch(err => res.status(500).json({ err: 'oops' }));
+});
+
+app.get('/posts', (req: Request, res: Response) => {
+  db.Post.findAll()
+    .then((posts: PostInstance[]) => res.status(200).json({ posts }));
+    .catch(err => res.status(500).json({ err: 'oops' }));
+});
+```
+
+Now, I encourage you to not copy & paste the above and write them yourself. Just stop and marvel at how every time you type, your text editor is able to tell you all the methods available, their types (in lots of detail!), and even documentation for each. Glorious.
```

Comment:
might need a small note to remind people to install the required extensions for this to happen (persp v2 readme has them listed I think)

Reply:
Ahhhh excellent point. I'll add this